src/bm_sim/dev_mgr.cpp

```diff
@@ -282,9 +282,10 @@ DevMgr::clear_port_stats(port_t port_num) {
 std::string
 DevMgr::sample_packet_data(const char *buffer, int len) {
   size_t amount = std::min(dump_packet_data, static_cast<size_t>(len));
-  assert(amount > 0);
   std::ostringstream ret;
-  utils::dump_hexstring(ret, &buffer[0], &buffer[amount]);
+  if (amount > 0) {
```

Comment:
IMO, it is usually more readable to have an early return:
```
if (amount == 0) {
        return "";
}
std::ostringstream ret;
...
```
it may be optimized away, but there is no need to instantiate `std::ostringstream ret` if amount is 0.

Reply:
Thanks as always for the explicit review comments, that even a C++ dilettante like me can follow :-)